src/main/resources/workflows/0.2.1/irida_workflow.xml

```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<iridaWorkflow>
+  <id>17fcb9f1-9501-4530-b45b-6d0673a368d1</id>
+  <name>amr-detection</name>
+  <version>0.2.1</version>
+  <analysisType>AMR_DETECTION</analysisType>
+  <inputs>
+    <sequenceReadsPaired>sequence_reads_paired</sequenceReadsPaired>
+    <requiresSingleSample>true</requiresSingleSample>
+  </inputs>
+  <parameters>
```

Comment:
Instead of requiring a fixed database name of `amr_detection` you could add a `<dynamicSource>` parameter here.  This would make the specific database used by RGI configurable as a parameter in IRIDA, with the options pre-loaded based on what's available in Galaxy. This means that you can select from multiple RGI databases in IRIDA and aren't restricted to just one.

This is described in https://phac-nml.github.io/irida-documentation/developer/tools/pipelines/workflow-description/#dynamicsource and would look something like:

```xml
<parameter name="rgi_database" required="true">
    <dynamicSource>
        <galaxyToolDataTable name="rgi_databases" displayColumn="name" parameterColumn="value" />
    </dynamicSource>
    <toolParameter toolId="toolshed.g2.bx.psu.edu/repos/card/rgi/rgi/5.1.1"
        parameterName="rgi_db_local" />
</parameter>
```

Reply:
Sure, I'll try and paste that in.

Reply:
Does IRIDA implicitly handle the tool conditional block to enable that parameter?

Reply:
Do you mean if you use the above code to set the local database parameter, will it set the option to use a **Locally installed RGI database**?

![image](https://user-images.githubusercontent.com/200517/128744429-fdcc6e56-065e-4329-bc2e-e4fd2a2fdea0.png)

No, it will not. Though if you have that set by default in the workflow it should all function properly.

Reply:
It would be nice if it would work with both **Default RGI database** or **Locally installed RGI database**, but I don't know how easy it would be to represent this to a user in IRIDA to make it obvious what needs to be changed.

Reply:
I have a PR pending that will restore the default db functionality: https://github.com/arpcard/rgi_wrapper/pull/16
The larger issue is that it isn't good for reproducibility if the database is different for each IRIDA installation.
Setting it to a local database at least allows the admin to specify the version of database to use, which is hopefully coordinated in the community.

Reply:
Thanks for linking to that other PR.

I do think it is a good idea to have the RGI database configurable in Galaxy, so including the changes you have made makes sense.

I do believe that reproducibility is not lost in the existing tool though as I believe the RGI database is stored within the conda package and linked to a specific version of RGI. That is to say, installing a specific version of RGI in Galaxy will always give you the same database version as the default database. Hence, all IRIDA installations should already be using the same RGI database so long as they are using the same RGI tool version.

Reply:
Ah, you are right. 'database' as referred to in the code is not the CARD database but the blast/diamond database.